### PR TITLE
[chassis] Fix test_bgp_update_timer for T2 topologies

### LINKED ISSUE #5176: test_bgp_update_timer fails in T2 topology
**Description**
The `test_bgp_update_timer` fails with below error
```
  File "/usr/local/lib/python2.7/dist-packages/_pytest/python.py", line 1464, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/usr/local/lib/python2.7/dist-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 81, in get_result
    _reraise(*ex)  # noqa
  File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/lib/python2.7/dist-packages/_pytest/python.py", line 174, in pytest_pyfunc_call
    testfunction(**testargs)
  File "/azp/agent/_work/6/s/tests/bgp/test_bgp_update_timer.py", line 207, in test_bgp_update_timer
    assert bgp_facts["bgp_neighbors"][n0.ip]["state"] == "established"
AssertionError

```

**Steps to reproduce the issue:**
1.
2.
3.

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important:**

    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

### PR DESCRIPTION
**Pull request by arlakshm**
Signed-off-by: Arvindsrinivasan Lakshminarasimhan <arlakshm@microsoft.com>

### Description of PR

Summary:
Fixes #5176

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x ] Test case(new/improvement)

### Back port request
- [ ] 201911

### Approach
Update the `test_bgp_update_timer` to run on T2 topologies

#### How did you do it?
The following changes are done.
- Change the test and conftest to use `enum_rand_one_per_hwsku_frontend_hostname`, to avoid the test from running on supervisor
- Change the subnet used to setup bgp session from `10.0.0.0/24` to `20.0.0.0/24`
- use `minigraph_ptf_indices` to get the correct PTF interfaces in the t2 topology.

#### How did you verify/test it?
Run the `test_bgp_update_timer` on T2 topology
```
arlakshm@2ec4d0fd85f8:/data/repos/my_mgmt/tests$ ./run_tests.sh -c bgp/test_bgp_update_timer.py -i '../ansible/str2,../ansible/veos'  -n vms26-t2-sonic-1  -u -e '--skip_sanity'
=== Running tests in groups ===
/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.exceptions import InvalidSignature
============================================================================================================================ test session starts ============================================================================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.10.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /data/repos/my_mgmt/tests, inifile: pytest.ini
plugins: forked-1.3.0, metadata-1.11.0, xdist-1.28.0, html-1.22.1, repeat-0.9.1, ansible-2.2.2
collecting ... /usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.exceptions import InvalidSignature
collected 2 items

bgp/test_bgp_update_timer.py::test_bgp_update_timer[str2-sonic-lc6-1] PASSED                                                                                                                                                                                           [ 50%]
bgp/test_bgp_update_timer.py::test_bgp_update_timer[str2-sonic-lc7-1] PASSED                                                                                                                                                                                           [100%]

--------------------------------------------------------------------------------------------------------- generated xml file: /data/repos/my_mgmt/tests/logs/tr.xml ---------------------------------------------------------------------------------------------------------
======================================================================================================================== 2 passed in 487.44 seconds =========================================================================================================================
INFO:root:Can not get Allure report URL. Please check logs
arlakshm@2ec4d0fd85f8:/data/repos/my_mgmt/tests$
```

### CODE REVIEW

File: tests/bgp/conftest.py [hunk 1]
```diff
@@ -353,8 +353,13 @@ def _setup_interfaces_t1_or_t2(mg_facts, peer_count):
...
 
             subnet_prefixlen = list(used_subnets)[0].prefixlen
-            _subnets = ipaddress.ip_network(u"10.0.0.0/24").subnets(new_prefix=subnet_prefixlen)
-            subnets = (_ for _ in _subnets if _ not in used_subnets)
+            if "t2" in tbinfo["topo"]["name"]:
+                # in T2 topologies the IP subnet could conflict with interfaces on other linecards. 
+                # use a complete different subnet
+                subnets = ipaddress.ip_network(u"20.0.0.0/24").subnets(new_prefix=subnet_prefixlen)
+            else:
+                _subnets = ipaddress.ip_network(u"10.0.0.0/24").subnets(new_prefix=subnet_prefixlen)
```
Comment (smaheshm) [inline tests/bgp/conftest.py:361]: why not use 20.0.0.0 in both cases.

**Comment by abdosi** [comment]
LGTM other than above comment

**Review by abdosi — APPROVED** [review]

**Comment by wenyiz2021** [comment]
Requested for backporting to 202012 test branch.
The same issue was seen on a single asic dualtor, and this PR fixes that.

**Pull request merged**